src/app/app.component.ts

```diff
@@ -24,26 +22,8 @@ export class AppComponent implements OnInit {
     }
   }
 
-  openMenu() {
-    this.menuOpen = !this.menuOpen;
-  }
-
-  collapseOpen() {
-    this.collapsed = !this.collapsed;
-  }
-
-  closeMenu() {
-    this.menuOpen = false;
-  }
-
-  logout() {
-    this.userService.logout();
-    this.router.navigateByUrl('/login').then(() => {
-      this.closeMenu();
-    });
-  }
-
-  onSearch(query: string) {
-    this.router.navigate(['/search', query]);
+  // Need to export the side-menu function to a dedicated service
+  toggleSideMenu() {
+    this.sideMenuClosed = !this.sideMenuClosed;
   }
```

Comment:
@bfkelsey What do you think of this comment ?

I think we should move the few function that toggle the side menu to a dedicated service. So whe can read the state of the side menu if open or not, toggle it from anywhere in the app, and also maybe store it into cookies / user settings  (maybe for future trigger toggle on hover ?) ?

So I'm trying to get it working but no success, could you point me to a guide that integrates two way bindings for a boolean ? This is the basics but I don't understand why I can't set a boolean as Observable ?

Reply:
Nevermind found a way to get it working !